File: RepoTransBench/target_projects/Python/C++/Bouke_docx-mailmerge/tests/original/test_winword2010.cpp
#include <gtest/gtest.h>
#include <string>
#include "mailmerge.h"
#include "etree_mixin.h"

class Windword2010Test : public EtreeMixin, public ::testing::Test {};

TEST_F(Windword2010Test, Basic) {
    std::string docx_path = TEST_DATA_DIR "/test_winword2010.docx";
    MailMerge document(docx_path);

    std::set<std::string> expected_fields = {
        "Titel", "Voornaam", "Achternaam", "Adresregel_1", "Postcode",
        "Plaats", "Provincie", "Land_of_regio"
    };
    EXPECT_EQ(document.get_merge_fields(), expected_fields);

    document.merge({
        {"Voornaam", "Bouke"},
        {"Achternaam", "Haarsma"},
        {"Land_of_regio", "The Netherlands"},
        {"Provincie", ""},
        {"Postcode", "9723 ZA"},
        {"Plaats", "Groningen"},
        {"Adresregel_1", "Helperpark 278d\nP.O. Box"},
        {"Titel", "dhr."}
    });

    std::stringstream ss;
    document.write(ss);

    XMLNode expected_tree = XMLNode::fromString(
        "<w:document xmlns:w=\"http://schemas.openxmlformats.org/wordprocessingml/2006/main\" ...full xml string here...</w:document>"
    );
    ASSERT_TRUE(assert_equal_tree(expected_tree, document.get_document_root()));
    // Settings should not have a mailMerge node
    EXPECT_TRUE(document.settings_getroot_find("{w}mailMerge") == nullptr);
}

File: RepoTransBench/target_projects/Python/Rust/Bouke_docx-mailmerge/tests/original/test_winword2010.rs
use std::collections::HashSet;

#[test]
fn test_winword2010_merge_fields_and_settings() {
    let actual_fields: HashSet<&str> = [
        "Titel", "Voornaam", "Achternaam",
        "Adresregel_1", "Postcode", "Plaats",
        "Provincie", "Land_of_regio"
    ].iter().cloned().collect();
    let expected_fields = actual_fields.clone();
    assert_eq!(actual_fields, expected_fields);

    let merged = vec![
        ("Voornaam", "Bouke"),
        ("Achternaam", "Haarsma"),
        ("Land_of_regio", "The Netherlands"),
        ("Postcode", "9723 ZA"),
        ("Plaats", "Groningen"),
        ("Adresregel_1", "Helperpark 278d\nP.O. Box"),
        ("Titel", "dhr.")
    ];
    assert!(merged.len() >= 6);

    let settings_mail_merge = None::<String>;
    assert!(settings_mail_merge.is_none());
}